backend/news-service/src/utils.js

```diff
@@ -0,0 +1,79 @@
+function compareDates(a, b) {
+  const dA = new Date(a.from);
+  const dB = new Date(b.from);
+  if (dA < dB) {
+    return -1;
+  }
+  if (dA > dB) {
+    return 1;
+  }
+  return 0;
+}
+
+function getNextDay(date) {
+  const nextDay = new Date(date);
+  nextDay.setDate(nextDay.getDate() + 1);
+  return nextDay;
+}
+
+function getPreviousDay(date) {
+  const nextDay = new Date(date);
+  nextDay.setDate(nextDay.getDate() - 1);
+  return nextDay;
+}
+
+function getIncludedTimespans(newFrom, newUntil, timespans) {
+  let tentFrom = null;
+  let tentUntil = null;
+  let placedFrom = false;
+  const includedTimespans = [];
+  for (let i = 0; i < timespans.length; i += 1) {
+    tentFrom = new Date(timespans[i].from);
+    tentUntil = new Date(timespans[i].until);
+    if (!placedFrom) {
+      if (newFrom > getNextDay(tentUntil)) {
```

Comment:
Could be combined with the statement above.

Reply:
Go away